src/android/OpenTokAndroidPlugin.java

```diff
@@ -492,9 +504,31 @@ public void run() {
                 mSession.sendSignal(args.getString(0), args.getString(1), c);
             }
         } else if (action.equals("unpublish")) {
-
+            Log.i( TAG, "unpublish command called");
+            if (myPublisher != null) {
+                cordova.getActivity().runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        myPublisher.stopPublishing();
```

Comment:
@mark-veenstra The `publisher` does not have a method that stops it from publishing. You have to use the `unpublish` method from the session.

Reply:
@mark-veenstra my mistake, I didn't see the `stopPublishing` method. Please ignore first comment

Reply:
@msach22 yeah the stopPublishing method was the existing one